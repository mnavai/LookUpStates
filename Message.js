import React from 'react';
import './Message.css';

export default function Message(props) {
    const badge = +props.unread 
      ? <div className="unread_count badge">{props.unread}</div>
      : null;
    return (
      <div className="Message">
          {props.name}
          {badge}
      </div>

 );
}