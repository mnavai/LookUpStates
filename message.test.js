import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Message from './Message';

describe('Message component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Message />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Message name="Message" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no unreads', () => {
    const tree = renderer
      .create(<Message name="Message" unread={0}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});