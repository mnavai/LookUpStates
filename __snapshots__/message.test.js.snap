// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders the UI as expected 1`] = `
<div
  className="Message"
>
  Message
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`Message component renders the UI as expected with no unreads 1`] = `
<div
  className="Message"
>
  Message
</div>
`;
